### ISSUE TITLE
in project configuration only the first channel works

### ISSUE THREAD

**Issue by refextu** [jenkinsci/slack-plugin#98]
Added to the channel configuration:

#builds, #live
or
#builds, #staging 

Expected: Notification send to both channels.
current situation: Notification is only send to first channel.

**Comment by philowest** [comment]
So this issue was introduced with the changes for the test connection functionality. Working on a fix now.

**Comment by samrocketman** [comment]
Cool, thanks a lot for the quick fix @philowest.  When I merge all the latest patches I'll generate a patch release.

**Comment by samrocketman** [comment]
Merged the PR after successful testing.  Look forward to 2.0.

**Comment by q-a** [comment]
I'm seeing the same issue.

**Comment by samrocketman** [comment]
Yep, this was confirmed and fixed in master.  It will be available in the next release.

**Comment by vanga** [comment]
can we also send notifications to channels in multiple domains?
Comma separated list doesn't work on version 1.8?
Will it work with new releases? is it on the roadmap?

Thanks.

**Comment by samrocketman** [comment]
As noted in the linked pull request this will be fixed.  You can message more than one domain in a Jenkins instance but _not_ more than one in a single job.

Posting to more than one domain in a single job could work if #41 was fixed.

I welcome you to contribute.  I don't develop this plugin.  I only maintain pull requests and releases.  The project is shaped by contributors so there is no documented roadmap unless you consider the groomed list of issues a roadmap.

**Comment by SB0-Lee** [comment]
@samrocketman Any idea when this one will deploy?

**Comment by samrocketman** [comment]
Hi @SB0-Lee, this will not be deployed until #106 is resolved.  I won't deploy the upgrade without a migration path for existing users.

I don't actively develop this plugin.  I primarily manage pull requests and releases.

**Comment by Nooba** [comment]
Hi @samrocketman 

Any chance you could just ship this fix in a 1.8.1 version ?
It would be really nice not to freeze the updates because you are expecting to release a new major version if possible.

Thanks in advance,
Regards

**Comment by samrocketman** [comment]
That might be an option.  I'll see if I can cleanly cherry-pick the changes.  Obviously, the desired result is for someone to contribute the migration path in #106.  But in the mean time a short-term fix isn't terrible.  Though, I'd rather have the former.  I only maintain pull requests and releases so it's largely up to contributors to focus on how the plugin progresses.

**Comment by Nooba** [comment]
Unsure whether this might help you (especially since I rewrote history), but I forked the repo and applied every PR except for post-build action one. This one I left on a feature branch, as it should imho wait here as long as no migration plan is found.
https://github.com/Nooba/slack-plugin
Please note that the changelog is to be done properly, and I haven't actually tested any of this, and although the conflict were rather easy to solve, It should be tested if you decide to go from there.
Thanks.

**Comment by samrocketman** [comment]
I'd actually, rather the earnest of people wanting an update prioritize the migration being worked on.  Shoving the post-build action onto a feature branch I don't think would help the project move forward.  I might look to your repository as a reference if I do a point release for the 1.8 series but I likely won't take any history that's been rewritten.  It may just be a 1.8-stable branch.  For the most part, once _introduced regressions_ are fixed I'll refuse to update the 1.8 series until the migration is addressed by someone.  Sweeping it under the rug isn't going to help.

**Comment by bootstraponline** [comment]
For what it's worth, when I contacted Slack about this issue a while ago the response was that so few of their customers use the Jenkins plugin that they have no plans to contribute bug fixes.

If you have a paid Slack account then I suggest opening a support ticket asking them to reconsider supporting this plugin.

**Comment by samrocketman** [comment]
I'd also like to add to @bootstraponline's statement that I don't actually use the Slack plugin myself.  I continue to maintain the plugin (for pull requests and releases only) because there was a time when there was no maintainer at all.  Contributors were getting frustrated that they couldn't get their pull requests accepted or a release out.  So I picked up maintaining the plugin in a "pull request only" relationship with the project.  That's why I don't actively develop this plugin myself (for the most part).

I'll continue to maintain this plugin because I feel I've got a good process down for code review and accepting pull requests by this point.  Nothing is stopping someone from contributing a fix for the migration issue.
